streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -777,4 +778,23 @@ public synchronized TopologyDescription describe() {
         return internalTopologyBuilder.describe();
     }
 
+    // Adjust the generated topology based on the configs.
+    // Not exposed as public API and should be removed post 2.0
+    Topology adjust(final StreamsConfig config) {
+        final boolean enableOptimization20 = config.getString(StreamsConfig.TOPOLOGY_OPTIMIZATION).equals(StreamsConfig.OPTIMIZE_AT_20);
+
+        if (enableOptimization20) {
+            for (Map.Entry<StoreBuilder, String> entry : internalTopologyBuilder.storeToSourceChangelogTopic.entrySet()) {
+                final StoreBuilder storeBuilder = entry.getKey();
+                final String topicName = entry.getValue();
+
+                // update store map to disable logging for this store
+                storeBuilder.withLoggingDisabled();
+                internalTopologyBuilder.addStateStore(storeBuilder, true);
```

Comment:
why do we need to add the store again? Isn't `storeBuilder` the same object that is already present and we overwrite it with itself here? `storeBuilder.withLoggingDisabled()` is a mutable call, right?

Reply:
Unfortunately not.. note that `addStateStore` directly constructs the `StoreFactory` and put that into the map `stateFactories`. And `StoreFactory` is immutable. So I have to overwrite the key with a new StoreFactory in this way.

Reply:
Ack.